### PR TITLE
Leaderboard

### PR DESCRIPTION
**Pull request by ErikBoonprakong**
Added win/lose logic to playOnline to automatically update the leaderboard

### CODE REVIEW

File: chess/src/Components/Leaderboard.js [hunk 1]
```diff
@@ -28,11 +23,11 @@ class Leaderboard extends React.Component {
   }
 
   render() {
-    console.log(this.state.data.leaderboard);
-    const leaderboard = this.state.data.leaderboard.map((person) => {
+    console.log(this.state.data);
+    const leaderboard = this.state.data.leaderboard.map((person, i) => {
```
Comment (megwat) [inline chess/src/Components/Leaderboard.js:27]: maybe this could go in a separate function?

File: chess/src/Components/PlayOnline.js [hunk 2]
```diff
@@ -18,20 +16,80 @@ class PlayOnline extends React.Component {
...
+      this.setState({ players: playerList });
+    });
+    this.socket.on("leave room", (playerList) => {
+      this.setState({ players: playerList });
+    });
+    // if (this.state.players.length < 2) {
+    //   console.log("no players");
+    //   this.socket.emit(
+    //     "new move",
+    //     "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1"
```
Comment (megwat) [inline chess/src/Components/PlayOnline.js:45]: is this commented out code needed

File: chess/src/Components/PlayOnline.js [hunk 3]
```diff
@@ -48,50 +106,51 @@ class PlayOnline extends React.Component {
...
 
-  async sendResults(won, lost, draw) {
-    const response = await this.networking.postResult(
-      this.currentUser.user_id,
-      this.currentUser.username,
-      won,
-      lost,
-      draw
-    );
+  async sendResults(name, won, lost, draw) {
```
Comment (megwat) [inline chess/src/Components/PlayOnline.js:147]: good rework of the sendResults function

**Review by megwat — APPROVED** [review]

File: chess/src/Components/Leaderboard.js (hunk 1 above)
Comment (ErikBoonprakong) [reply chess/src/Components/Leaderboard.js:25]: good idea

**Pull request merged**